### PR TITLE
Remove traitgen from OpenAssetIO codebase.

### LINKED ISSUE #715: Remove codegen from OpenAssetIO repo
## What

Delete traitgen from the OpenAssetIO repo

## ACs

- All references removed from the OpenAssetIO build system/codebase
- Doc references updated(?)
- OpenAssetIO repo CI integration check (integration.yml)

## Notes

- Integration test follow similar pattern to BAL and simpleResolver (checkout external repo and run its `pytest`s).

### PR DESCRIPTION
**Pull request by elliotcmorris**
Closes #715 

Two commits here

1. Removes traitgen from the OpenAssetIO codebase, along with all requirements and test targets.
2. Adds an integration test from the new traitgen repo.

### CODE REVIEW

**Comment by feltech** [comment]
[consider] Minor commit message note: it's good to include the issue number in all the commit message bodies for later archaeology (sometimes we don't if there's a good reason); and you don't need to put a full markdown hyperlink, just the #number.

**Review by feltech — APPROVED** [review]
LGTM. Also nice to have the `ctest` speedup (without having to explicitly disable traitgen).  Should also mitigate (though not neatly solve) #687

File: .github/workflows/integrations.yml [hunk 1]
```diff
@@ -60,6 +60,16 @@ jobs:
           repository: OpenAssetIO/OpenAssetIO-Manager-BAL
           path: external/BAL
 
+      - name: Checkout TraitGen
```
[the remark this answers is not in the document]
Comment (elliotcmorris) [reply .github/workflows/integrations.yml:73]: Yeah I think I agree.

I've reordered the other steps too so the steps are co-located, (BAL was the only other troublemaker)

[efd2a3b](https://github.com/OpenAssetIO/OpenAssetIO/pull/725/commits/efd2a3b9efb07b82a7e4c2a29bc35e960c25332e)

[the remark this answers is not in the document]
Comment (elliotcmorris) [reply .github/workflows/integrations.yml:73]: Oh right, of course, simpleresolver is dependent on BAL... oops.

**Review by feltech — APPROVED** [review]
LGTM squash and merge when ready

**Pull request merged**